### ISSUE TITLE
RuntimeError: database is locked

### ISSUE THREAD

**Issue by jeroenmaelbrancke** [openvstorage/alba-asdmanager#255]
Found the stacktrace in the ovs-webapp-api logging.

@khenderick do you need extra information or is this enough?

```
2017-06-20 15:14:44 20700 +0200 - NY1SRV0019 - 24078/140500974799280 - api/logger - 25416 - WARNING - API call albabackends.retrieve took 59.75s
2017-06-20 15:14:44 20700 +0200 - NY1SRV0019 - 24077/140501297238096 - api/logger - 20918 - WARNING - API call albabackends.retrieve took 119.36s
2017-06-20 15:15:44 91100 +0200 - NY1SRV0019 - 24078/140500974799280 - extensions/api - 25417 - INFO - [albabackends.retrieve] - a67b8755-5361-47ab-b14e-92af40a06d02 - [] - {"pk": "56f58646-419d-4236-a868-e3b79ac8784d"} - {"cookies": {}, "meta": {"HTTP_AUTHORIZATION": "Bearer 8YIdI:S?WzbmlGDx*]scz_p}Cu7Dh1T_?wN6s9rz/.uhA]l3b;*f8<K[hzYkd4Jb", "wsgi.multiprocess": "True", "HTTP_X_FORWARDED_SSL": "on", "SERVER_SOFTWARE": "gunicorn/19.4.5", "SCRIPT_NAME": "/api", "REQUEST_METHOD": "GET", "PATH_INFO": "/alba/backends/56f58646-419d-4236-a868-e3b79ac8784d/", "SERVER_PROTOCOL": "HTTP/1.0", "QUERY_STRING": "contents=local_summary%2Clive_status", "HTTP_X_REAL_IP": "172.17.16.19", "HTTP_USER_AGENT": "python-requests/2.9.1", "HTTP_CONNECTION": "close", "SERVER_NAME": "127.0.0.1", "REMOTE_PORT": "56148", "wsgi.url_scheme": "https", "SERVER_PORT": "8002", "HTTP_X_SCHEME": "https", "wsgi.input": "<gunicorn.http.body.Body object at 0x7fc8e6dea110>", "HTTP_HOST": "172.17.16.22", "wsgi.multithread": "True", "HTTP_ACCEPT": "application/json; version=3", "wsgi.version": "(1, 0)", "RAW_URI": "/alba/backends/56f58646-419d-4236-a868-e3b79ac8784d/?contents=local_summary%2Clive_status", "wsgi.run_once": "False", "wsgi.errors": "<gunicorn.http.wsgi.WSGIErrorsWrapper object at 0x7fc8e6deadd0>", "REMOTE_ADDR": "127.0.0.1", "gunicorn.socket": "<socket fileno=13 sock=127.0.0.1:8002 peer=127.0.0.1:56148>", "CONTENT_TYPE": "application/json", "wsgi.file_wrapper": "<class 'gunicorn.http.wsgi.FileWrapper'>", "CSRF_COOKIE": "cfhHYsTpDOPDio6KMc4Mf3UOvq4YjoTU", "HTTP_ACCEPT_ENCODING": "gzip, deflate"}, "request": {"contents": "local_summary,live_status"}}
2017-06-20 15:15:44 91100 +0200 - NY1SRV0019 - 24078/140500974799280 - log/api - 25417 - INFO - [albabackends.retrieve] - a67b8755-5361-47ab-b14e-92af40a06d02 - [] - {"pk": "56f58646-419d-4236-a868-e3b79ac8784d"} - {"cookies": {}, "meta": {"HTTP_AUTHORIZATION": "Bearer 8YIdI:S?WzbmlGDx*]scz_p}Cu7Dh1T_?wN6s9rz/.uhA]l3b;*f8<K[hzYkd4Jb", "wsgi.multiprocess": "True", "HTTP_X_FORWARDED_SSL": "on", "SERVER_SOFTWARE": "gunicorn/19.4.5", "SCRIPT_NAME": "/api", "REQUEST_METHOD": "GET", "PATH_INFO": "/alba/backends/56f58646-419d-4236-a868-e3b79ac8784d/", "SERVER_PROTOCOL": "HTTP/1.0", "QUERY_STRING": "contents=local_summary%2Clive_status", "HTTP_X_REAL_IP": "172.17.16.19", "HTTP_USER_AGENT": "python-requests/2.9.1", "HTTP_CONNECTION": "close", "SERVER_NAME": "127.0.0.1", "REMOTE_PORT": "56148", "wsgi.url_scheme": "https", "SERVER_PORT": "8002", "HTTP_X_SCHEME": "https", "wsgi.input": "<gunicorn.http.body.Body object at 0x7fc8e6dea110>", "HTTP_HOST": "172.17.16.22", "wsgi.multithread": "True", "HTTP_ACCEPT": "application/json; version=3", "wsgi.version": "(1, 0)", "RAW_URI": "/alba/backends/56f58646-419d-4236-a868-e3b79ac8784d/?contents=local_summary%2Clive_status", "wsgi.run_once": "False", "wsgi.errors": "<gunicorn.http.wsgi.WSGIErrorsWrapper object at 0x7fc8e6deadd0>", "REMOTE_ADDR": "127.0.0.1", "gunicorn.socket": "<socket fileno=13 sock=127.0.0.1:8002 peer=127.0.0.1:56148>", "CONTENT_TYPE": "application/json", "wsgi.file_wrapper": "<class 'gunicorn.http.wsgi.FileWrapper'>", "CSRF_COOKIE": "cfhHYsTpDOPDio6KMc4Mf3UOvq4YjoTU", "HTTP_ACCEPT_ENCODING": "gzip, deflate"}, "request": {"contents": "local_summary,live_status"}}
2017-06-20 15:15:47 13200 +0200 - NY1SRV0019 - 24078/140500943316304 - extensions/volatile mutex - 25418 - WARNING - A lock on ovs_lock_ovs_data_albanode_6a733278-d222-4342-814a-91bc61843b61_storage_stack was kept for 0.744675159454 sec
2017-06-20 15:15:54 16100 +0200 - NY1SRV0019 - 24078/140500943318864 - extensions/volatile mutex - 25419 - WARNING - A lock on ovs_lock_ovs_data_albanode_9a84dc90-4b0c-40d9-9ad4-b268a0b75238_storage_stack was kept for 7.73672389984 sec
Exception in thread Thread-102546:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 801, in __bootstrap_inner
    self.run()
  File "/usr/lib/python2.7/threading.py", line 754, in run
    self.__target(*self.__args, **self.__kwargs)
  File "/opt/OpenvStorage/ovs/dal/hybrids/albabackend.py", line 97, in _load_live_info
    _data = _node.storage_stack
  File "/opt/OpenvStorage/ovs/dal/dataobject.py", line 345, in <lambda>
    fget = lambda s: s._get_dynamic_property(dynamic)
  File "/opt/OpenvStorage/ovs/dal/dataobject.py", line 414, in _get_dynamic_property
    return self._backend_property(data_loader, dynamic)
  File "/opt/OpenvStorage/ovs/dal/dataobject.py", line 1011, in _backend_property
    dynamic_data = function()
  File "/opt/OpenvStorage/ovs/dal/hybrids/albanode.py", line 63, in _storage_stack
    disk_data = self.client.get_disks()
  File "/opt/OpenvStorage/ovs/extensions/plugins/asdmanager.py", line 137, in get_disks
    return self._call(requests.get, 'disks', clean=True)
  File "/opt/OpenvStorage/ovs/extensions/plugins/asdmanager.py", line 107, in _call
    raise RuntimeError(error_message)
RuntimeError: database is locked
```

**Comment by wimpers** [comment]
Is this related to the corrupt SQL lite ticket? #256

**Comment by khenderick** [comment]
@wimpers, no. Ticket #256 was not about SQLite being corrupt, but a python standard library file that was corrupt. And SQLite was one of the libraries that make use of this standard library.

**Comment by khenderick** [comment]
Most likely we have some kind of multi threading going on that tries to write concurrently to the database.

**Comment by wimpers** [comment]
@khenderick @jeroenmaelbrancke  what does this break? Which failure made this bug come to the surface?

**Comment by JeffreyDevloo** [comment]
This just breaks certain calls if they are executed concurrently
Retrying would have r esolved the issue

**Comment by khenderick** [comment]
I'll give this some more prio as this is causing issues.

**Comment by khenderick** [comment]
Fixed by openvstorage/framework-extensions#13 packaged in openvstorage-extensions-0.1.0-dev.1499689618.50c4025

**Comment by JeffreyDevloo** [comment]
Tackled on a different repo.
